### PR TITLE
Story point components

### PR DESCRIPTION
**Pull request by sedroche**
@JameelB @dimitraz

storypointer

There are a few todos in here which I will follow up on.
I would like to get this in so we can integrate with your work @JameelB @dimitraz

### CODE REVIEW

File: react-web/src/components/App.js [hunk 1]
```diff
@@ -5,30 +5,63 @@ import './App.css';
...
+    };
+
+    this.onSessionIdCreated = this.onSessionIdCreated.bind(this);
+    this.onUserNameCompleted = this.onUserNameCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    // TODO:
+    // Hack just to get some routing working.
+    const sessionId = window.location.pathname.split('session/')[1] || null;
```
Comment (JameelB) [inline react-web/src/components/App.js]: I think you should be able to get this from props. `this.props.location`

File: react-web/src/components/JoinSession.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+
+import { UsernameForm } from './UsernameForm';
+
+export class JoinSession extends React.Component {
```
Comment (JameelB) [inline react-web/src/components/JoinSession.js]: If there's no further logic involved here, i think you can make this as a functional component instead.

File: react-web/src/components/StoryPointer.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    }
+    return result;
+  }
+
+  createPointersAsShowing() {
+    return this.state.pointers.map(pointer => Object.assign({}, pointer, {show: true}));
+  }
+
+  createClearedPointers() {
+    return this.state.pointers.map(pointer => Object.assign({}, pointer, {score: ''}));
```
Comment (JameelB) [inline react-web/src/components/StoryPointer.js]: When the points are shown, there is no way of getting them set to hidden again. Can we include setting `show` to false here as well to reset it?

Comment (sedroche) [reply react-web/src/components/StoryPointer.js]: :+1:

File: react-web/src/components/JoinSession.js (hunk 2 above)
Comment (sedroche) [reply react-web/src/components/JoinSession.js]: Cool, thanks for pointing this out. I hadn't seen this pattern

**Pull request merged**